### PR TITLE
[opentelemetry-cpp] update version to 1.4.1

### PR DESCRIPTION
**Pull request by joaosoares**
**Describe the pull request**

- #### What does your PR fix?
  Update opentelemetry-cpp port to the latest version v1.4.1.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by joaosoares** [comment]
I'm not sure why, but the dependent package `azure-core-tracing-opentelemetry-cpp` is failing with the message

```
No transport adapter was selected, using libcurl as the default option for POSIX.
CMake Error at /Users/joao/Projects/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package):
  Could not find a configuration file for package "opentelemetry-cpp" that is
  compatible with requested version "1.3.0".

  The following configuration files were considered but not accepted:

    /Users/joao/Projects/vcpkg/installed/x64-osx/share/opentelemetry-cpp/opentelemetry-cpp-config.cmake, version: 1.4.1

Call Stack (most recent call first):
  CMakeLists.txt:39 (find_package)
```

Seems like it should work, as the version is greater than 1.3.0, but it doesn't.

**Comment by JonLiu1993** [comment]
> I'm not sure why, but the dependent package `azure-core-tracing-opentelemetry-cpp` is failing with the message
> 
> ```
> No transport adapter was selected, using libcurl as the default option for POSIX.
> CMake Error at /Users/joao/Projects/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package):
>   Could not find a configuration file for package "opentelemetry-cpp" that is
>   compatible with requested version "1.3.0".
> 
>   The following configuration files were considered but not accepted:
> 
>     /Users/joao/Projects/vcpkg/installed/x64-osx/share/opentelemetry-cpp/opentelemetry-cpp-config.cmake, version: 1.4.1
> 
> Call Stack (most recent call first):
>   CMakeLists.txt:39 (find_package)
> ```
> 
> Seems like it should work, as the version is greater than 1.3.0, but it doesn't.

@joaosoares, This error is because azure-core-tracing-opentelemetry-cpp set https://github.com/Azure/azure-sdk-for-cpp/blob/584f83c7bd0399a5c473508c459d3e11fcf2c764/sdk/core/azure-core-tracing-opentelemetry/CMakeLists.txt#L39
````
find_package(opentelemetry-cpp "1.3.0" CONFIG REQUIRED)
````
so when you update opentelemetry-cpp from "1.3.0" to "1.4.1" made 
````
  Could not find a configuration file for package "opentelemetry-cpp" that is
  compatible with requested version "1.3.0".
````
The soultion:
Port: azure-core-tracing-opentelemetry-cpp
vcpkg.json file:
````
    {
      "name": "opentelemetry-cpp",
      "version>=": "1.3.0"
    },
````
->
````
    {
      "name": "opentelemetry-cpp",
      "version>=": "1.4.1"
    },
````
Make a patch file:
file: https://github.com/Azure/azure-sdk-for-cpp/blob/584f83c7bd0399a5c473508c459d3e11fcf2c764/sdk/core/azure-core-tracing-opentelemetry/CMakeLists.txt#L39
````
find_package(opentelemetry-cpp "1.3.0" CONFIG REQUIRED)
````
->
````
find_package(opentelemetry-cpp "1.4.1" CONFIG REQUIRED)
````

**Comment by JonLiu1993** [comment]
> > I'm not sure why, but the dependent package `azure-core-tracing-opentelemetry-cpp` is failing with the message
> > ```
> > No transport adapter was selected, using libcurl as the default option for POSIX.
> > CMake Error at /Users/joao/Projects/vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package):
> >   Could not find a configuration file for package "opentelemetry-cpp" that is
> >   compatible with requested version "1.3.0".
> > 
> >   The following configuration files were considered but not accepted:
> > 
> >     /Users/joao/Projects/vcpkg/installed/x64-osx/share/opentelemetry-cpp/opentelemetry-cpp-config.cmake, version: 1.4.1
> > 
> > Call Stack (most recent call first):
> >   CMakeLists.txt:39 (find_package)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Seems like it should work, as the version is greater than 1.3.0, but it doesn't.
> 
> @joaosoares, This error is because azure-core-tracing-opentelemetry-cpp set https://github.com/Azure/azure-sdk-for-cpp/blob/584f83c7bd0399a5c473508c459d3e11fcf2c764/sdk/core/azure-core-tracing-opentelemetry/CMakeLists.txt#L39
> 
> ```
> find_package(opentelemetry-cpp "1.3.0" CONFIG REQUIRED)
> ```
> 
> so when you update opentelemetry-cpp from "1.3.0" to "1.4.1" made
> 
> ```
>   Could not find a configuration file for package "opentelemetry-cpp" that is
>   compatible with requested version "1.3.0".
> ```
> 
> The soultion: Port: azure-core-tracing-opentelemetry-cpp vcpkg.json file:
> 
> ```
>     {
>       "name": "opentelemetry-cpp",
>       "version>=": "1.3.0"
>     },
> ```
> 
> ->
> 
> ```
>     {
>       "name": "opentelemetry-cpp",
>       "version>=": "1.4.1"
>     },
> ```
> 
> Make a patch file: file: https://github.com/Azure/azure-sdk-for-cpp/blob/584f83c7bd0399a5c473508c459d3e11fcf2c764/sdk/core/azure-core-tracing-opentelemetry/CMakeLists.txt#L39
> 
> ```
> find_package(opentelemetry-cpp "1.3.0" CONFIG REQUIRED)
> ```
> 
> ->
> 
> ```
> find_package(opentelemetry-cpp "1.4.1" CONFIG REQUIRED)
> ```

@[LarryOsterman](https://github.com/LarryOsterman),Could you please look this plan feasible?

**Comment by joaosoares** [comment]
@JonLiu1993 patch added, all checks are passing now

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
- x64-linux

**Comment by BillyONeal** [comment]
I asked Azure folks to comment on the patch to their bits.

File: ports/azure-core-tracing-opentelemetry-cpp/use-opentelemetry-cpp-141.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/sdk/core/azure-core-tracing-opentelemetry/CMakeLists.txt b/sdk/core/azure-core-tracing-opentelemetry/CMakeLists.txt
+index bf3077e..2bd6d91 100644
+--- a/sdk/core/azure-core-tracing-opentelemetry/CMakeLists.txt
++++ b/sdk/core/azure-core-tracing-opentelemetry/CMakeLists.txt
+@@ -36,7 +36,7 @@ if (BUILD_AZURE_CORE_TRACING_OPENTELEMETRY)
+         find_package(azure-core-cpp REQUIRED)
+       endif()
+     endif()
+-    find_package(opentelemetry-cpp "1.3.0" CONFIG REQUIRED)
```
Comment (BillyONeal) [inline ports/azure-core-tracing-opentelemetry-cpp/use-opentelemetry-cpp-141.patch:9]: We should poke Azure folks to see if they can just drop the `"1.3.0"` part entirely.

File: ports/azure-core-tracing-opentelemetry-cpp/vcpkg.json [hunk 2]
```diff
@@ -16,7 +17,7 @@
     },
     {
       "name": "opentelemetry-cpp",
-      "version>=": "1.3.0"
+      "version>=": "1.4.1"
```
Comment (BillyONeal) [inline ports/azure-core-tracing-opentelemetry-cpp/vcpkg.json:20]: This isn't right, the minimum version Azure Core wants is still 1.3.0.

**Review by BillyONeal — CHANGES_REQUESTED** [review]

File: ports/azure-core-tracing-opentelemetry-cpp/use-opentelemetry-cpp-141.patch (hunk 1 above)
Comment (BillyONeal) [reply ports/azure-core-tracing-opentelemetry-cpp/use-opentelemetry-cpp-141.patch:9]: Please change this patch to be substantially identical to https://github.com/Azure/azure-sdk-for-cpp/pull/3844 and comment in `portfile.cmake` that's where it came from.

Comment (BillyONeal) [reply ports/azure-core-tracing-opentelemetry-cpp/use-opentelemetry-cpp-141.patch:9]: Sorry, to clarify:

https://github.com/Azure/azure-sdk-for-cpp/pull/3844.patch

**Comment by joaosoares** [comment]
Thank you, will implement these changes

File: ports/azure-core-tracing-opentelemetry-cpp/use-opentelemetry-cpp-141.patch (hunk 1 above)
Comment (joaosoares) [reply ports/azure-core-tracing-opentelemetry-cpp/use-opentelemetry-cpp-141.patch:9]: Updated the patch and added a comment with the proper reference

File: ports/azure-core-tracing-opentelemetry-cpp/vcpkg.json (hunk 2 above)
Comment (joaosoares) [reply ports/azure-core-tracing-opentelemetry-cpp/vcpkg.json]: Updated it back to 1.3.0

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**